internal/authorization/serviceaccount_authenticator.go

```diff
@@ -0,0 +1,178 @@
+	case len(saa.clientID) > 0 && len(saa.keyID) > 0 && len(saa.privateKeyData) > 0:
+		var key *rsa.PrivateKey
+		key, err = rsaKeyFromBase64(saa.privateKeyData)
+		if err != nil {
+			break
+		}
+		jwt, err = getJWTToken(context.Background(), saa.client, saa.keyID, saa.clientID, key)
+	default:
+		err = fmt.Errorf("inconsistent auth configuration")
+	}
+
+	if jwt != nil {
+		saa.userAuth.WriteJWTToken(jwt)
+	}
+
+	return jwt, err
+}
+
+func getClientCredentialsToken(ctx context.Context, apiClient client.Interface, clientID, clientSecret string) (*oauth2.Token, error) {
+	req := apiClient.Post().APIPath(serviceAccountAuthEndpoint)
+	endpoint := req.URL()
+
+	config := clientcredentials.Config{
+		ClientID:     clientID,
+		ClientSecret: clientSecret,
+		TokenURL:     endpoint.String(),
+		AuthStyle:    oauth2.AuthStyleInHeader,
+	}
+
+	tokenContext := context.WithValue(ctx, oauth2.HTTPClient, apiClient.HTTPClient())
+	return config.Token(tokenContext)
+}
+
+func getJWTToken(ctx context.Context, apiClient client.Interface, keyID, clientID string, key *rsa.PrivateKey) (*oauth2.Token, error) {
+	jwsHeader := &jws.Header{
+		Typ:       "JWT",
+		Algorithm: "RS256",
+		KeyID:     keyID,
+	}
+
+	jwsClaim := &jws.ClaimSet{
+		Iss: clientID,
+		Sub: clientID,
+		Aud: "console-client-credentials",
+		PrivateClaims: map[string]interface{}{
+			"jti": uuid.New(),
+		},
+	}
+
+	signedJWS, err := jws.Encode(jwsHeader, jwsClaim, key)
+	if err != nil {
+		return nil, err
+	}
+
+	values := make(url.Values)
+	values.Set("grant_type", "client_credentials")
+	values.Set("client_assertion_type", "urn:ietf:params:oauth:client-assertion-type:jwt-bearer")
+	values.Set("client_assertion", signedJWS)
+	values.Set("client_id", clientID)
+	values.Set("token_endpoint_auth_method", "private_key_jwt")
+	response, err := apiClient.
+		Post().
+		APIPath(serviceAccountAuthEndpoint).
+		SetHeader("Content-Type", "application/x-www-form-urlencoded").
+		Body([]byte(values.Encode())).
+		Do(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if response.Error() != nil {
+		return nil, response.Error()
+	}
+
+	type tempToken struct {
+		AccessToken string `json:"access_token"` //nolint: tagliatelle
+		TokenTipe   string `json:"token_type"`   //nolint: tagliatelle
+		ExpiresIn   int    `json:"expires_in"`   //nolint: tagliatelle
+	}
+
+	jwtToken := new(tempToken)
+
+	err = response.ParseResponse(jwtToken)
+	if err != nil {
+		return nil, err
+	}
+
+	return &oauth2.Token{
+		AccessToken: jwtToken.AccessToken,
+		TokenType:   jwtToken.TokenTipe,
+		Expiry:      time.Now().Add(time.Duration(jwtToken.ExpiresIn) * time.Second),
+	}, nil
+}
+
+func rsaKeyFromBase64(base64Data string) (*rsa.PrivateKey, error) {
+	keyData, err := base64.StdEncoding.DecodeString(base64Data)
+	if err != nil {
+		return nil, err
+	}
+
+	pemData, _ := pem.Decode(keyData)
+	key, err := x509.ParsePKCS8PrivateKey(pemData.Bytes)
+	return key.(*rsa.PrivateKey), err
```

Comment:
This type assertion should be handled with the second paeans (to verify it's ok). 
If the type is not correct, it will panic otherwise